packages/frontend-2/pages/projects/[id]/webhooks/index.vue

```diff
@@ -0,0 +1,269 @@
+      <HeaderNavLink
+        :to="`${projectRoute(projectId)}/webhooks`"
+        name="Webhooks"
+      ></HeaderNavLink>
+    </Portal>
+    <div class="flex items-center justify-between">
+      <h1 class="text-2xl font-bold">Webhooks</h1>
+      <div class="flex gap-2">
+        <FormButton color="secondary" :icon-left="BookOpenIcon" @click="openDocs">
+          Open Docs
+        </FormButton>
+        <FormButton :icon-left="PlusIcon" @click="showNewWebhookDialog = true">
+          Create Webhook
+        </FormButton>
+      </div>
+    </div>
+    <div class="my-8 text-sm">
+      Webhooks allow you to subscribe to a stream's events and get notified of them in
+      real time. You can then use this to trigger ci apps, automation workflows, and
+      more.
+    </div>
+
+    <LayoutTable
+      :columns="[
+        { id: 'enabled', header: 'State', classes: 'col-span-1' },
+        { id: 'data', header: 'Data', classes: 'col-span-5' },
+        {
+          id: 'triggers',
+          header: 'Trigger Events',
+          classes: 'col-span-6 whitespace-break-spaces text-xs'
+        }
+      ]"
+      :items="webhooks"
+      :buttons="[
+        {
+          icon: PencilIcon,
+          label: 'Edit',
+          action: openEditWebhookDialog,
+          class: 'text-primary'
+        },
+        {
+          icon: TrashIcon,
+          label: 'Delete',
+          action: openDeleteWebhookDialog,
+          class: 'text-red-500'
+        }
+      ]"
+    >
+      <template #enabled="{ item }">
+        <FormSwitch
+          :model-value="(item.enabled as boolean)"
+          @update:model-value="(newValue) => onChange(item, newValue)"
+        />
+      </template>
+      <template #data="{ item }">
+        <div class="flex flex-col">
+          <h3 class="font-bold text-base truncate">{{ item.description }}</h3>
+          <div class="flex gap-1.5 items-center">
+            <div class="h-4 w-4">
+              <InformationCircleIcon
+                v-if="getHistoryStatus(item) === 'noEvents'"
+                class="opacity-40"
+              />
+              <CheckCircleIcon
+                v-if="getHistoryStatus(item) === 'called'"
+                class="text-success"
+              />
+              <XCircleIcon
+                v-if="
+                  getHistoryStatus(item) === 'error' ||
+                  getHistoryStatus(item) === 'alert'
+                "
+                class="text-danger"
+              />
+            </div>
+            <span class="text-foreground opacity-50 text-sm truncate">
+              {{ getHistoryStatusInfo(item) }}
+            </span>
+          </div>
+          <span class="text-foreground opacity-50 text-sm truncate">
+            {{ item.url }}
+          </span>
+        </div>
+      </template>
+
+      <template #triggers="{ item }">
+        <div :class="{ 'opacity-60': !item.enabled }">
+          {{
+            (item.triggers as string[])
```

Comment:
I've adjusted the backend to no longer allow the array to contain `null` values so this should no longer be necessary